### ISSUE TITLE
"is not working as expected" on Ventura 13.5.1

### ISSUE THREAD

**Issue by lidnele** [Free-GPGMail/Free-GPGMail#95]
Hi,

When composing a mail, both the encryption and sign buttons are greyed out and sometimes a popup dialog appears, saying that "is not working as expected". When attempting to read encrypted mails, the same occurs. 

The details of the popup is:
"Decrypting the message failed.

Please re-install  from https://github.com/Free-GPGMail/Free-GPGMail.

Should this problem persist, contact us at https://github.com/Free-GPGMail/Free-GPGMail/issues and include the information that the xpc service is not working properly."

which is why I'm here now.

I've used free-gpgmail for many years and I use it through homebrew. As others I originally had the crash issue in 13.5 when composing mails. A free-gpgmail fix was issued and it stopped crashing. I'm 97 % sure I also tested that it actually worked at this time. I upgraded to 13.5.1 last week and have just now noticed that it doesn't work anymore.

I have tried uninstalling and reinstalling, disabling and enabling plugin (and accepting access rights) in Mail.app and so on to no avail.

Is there a log entry where I can obtain more information?

One potential cause may be that Mail.app has never found plugins in my homedir. I always have to manually copy them to /Library/Mail/Bundles/ after installation with homebrew. This has always been the case and has always worked. Until now, at least.

**Comment by bnavigator** [comment]
First thing I would do in your case is check if gpg from the suite is properly installed and can operate from the command line.

**Comment by lidnele** [comment]
Thank you for the reply. This seems to be working just fine. I tried these commands:

gpg --encrypt --sign --armor -r someemail@gmail.com -r my@email.com someFile and

gpg someFile.asc

and they encrypted and decrypted without issue.

Any idea what I should look at next?

**Comment by lidnele** [comment]
And for good measure, since I've already whipped out the terminal:

√ ~ % gpg --version                                                                                                  13:14:57
gpg (GnuPG/MacGPG2) 2.2.41
libgcrypt 1.8.10
Copyright (C) 2022 g10 Code GmbH
License GNU GPL-3.0-or-later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: /Volumes/Data/Users/elendil/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2
√ ~ %

**Comment by bnavigator** [comment]
Does your GPG Keychain show a proper set of keys, including your own private one?
Is your Libmacgpg up to date?

**Comment by lidnele** [comment]
Yes, the GPG keychain shows both keys as they should be. It's also what was used for the gpg commands earlier, I think (at least it complained, when I spelled one of the emails wrong).

How do I check libmacgpg? It's part of free-gpgmail, isn't it? As I mentioned I'm sure it worked before I upgrade from 13.5 to 13.5.1 - not that it's necessarily connected, and I've tried reinstalling the most recent free-gpgmail from brew.

**Comment by lidnele** [comment]
A bit more information. When replying to a mail (or when creating a new one), sometimes this pops up:
![image](https://github.com/Free-GPGMail/Free-GPGMail/assets/116806264/699f7c02-691b-4ac1-bb28-294a64f6d8f5)

Even though a recipient is specified AND encryption or signing is NOT enabled (nor should it be - it's just non-secure mails).

**Comment by bnavigator** [comment]
libmacgpg is used in the source of free-gpgmail but the compiled framework comes from the suite. Have you tried reinstalling the GPG Suite (`gpg-suite-no-mail` from homebrew or directly from the downloads page here or from GPGTools)?

**Comment by lidnele** [comment]
Thank you again for your swift reply.

I tried installing gpg-suite-no-mail (overwriting the existing installation, which was done with the "official" gpg-suite installer) with no effect. I also tried rebooting - I had to anyway, since brew hung on sudo access (which I googled could be solved by a reboot).

I then uninstalled both gpg-suite-no-mail and free-gpgmail, reinstalled them and rebooted once more and lo and behold, it works again.

It's a bit unsatisfactory not really knowing why, but I'm happy it works and sorry for using your time.

Though, upon reinstall of gpg-suite-no-mail, Ventura showed a bunch of popups of items added as background tasks. I wonder if one of these could somehow have been deactivated during the Ventura update. I didn't think to check when the problem was present.
I do wonder why fixGpgHome, GPG suite and shutdown-gpg-agent need to run in the background. The first and third, maybe, but the suite itself? Right now I don't need it _not_ working again, so I'll not experiment too much at the moment.

Thanks again.
